runtime/gc_staccato/StaccatoGC.cpp

```diff
@@ -231,127 +168,27 @@ MM_StaccatoGC::doTracing(MM_EnvironmentRealtime *env)
-					while(NULL != (clazz = classHeapIterator.nextClass())) {
-						if((0 == (J9CLASS_EXTENDED_FLAGS(clazz) & J9ClassGCScanned)) && _markingScheme->isMarked(clazz->classObject)) {
-							J9CLASS_EXTENDED_FLAGS_SET(clazz, J9ClassGCScanned);
-
-							/* Scan class */
-							GC_ClassIterator objectSlotIterator(env, clazz);
-							volatile j9object_t *objectSlotPtr = NULL;
-							while((objectSlotPtr = objectSlotIterator.nextSlot()) != NULL) {
-								didWork |= _markingScheme->markObject(env, *objectSlotPtr);
-							}
-
-							GC_ClassIteratorClassSlots classSlotIterator(clazz);
-							J9Class **classSlotPtr;
-							while((classSlotPtr = classSlotIterator.nextSlot()) != NULL) {
-								didWork |= _markingScheme->markClass(env, *classSlotPtr);
-							}
-						}
-					}
-					condYield(env, 0);
-				}
-			} else {
-				/* Check if the class loader has not been scanned but the class loader is live */
-				if( !(classLoader->gcFlags & J9_GC_CLASS_LOADER_SCANNED) && _markingScheme->isMarked((J9Object *)classLoader->classLoaderObject)) {
-					/* Flag the class loader as being scanned */
-					classLoader->gcFlags |= J9_GC_CLASS_LOADER_SCANNED;
-
-					GC_ClassLoaderSegmentIterator segmentIterator(classLoader, MEMORY_TYPE_RAM_CLASS);
-					J9MemorySegment *segment = NULL;
-					J9Class *clazz;
-
-					while(NULL != (segment = segmentIterator.nextSegment())) {
-						GC_ClassHeapIterator classHeapIterator(_javaVM, segment);
-						while(NULL != (clazz = classHeapIterator.nextClass())) {
-							/* Scan class */
-							GC_ClassIterator objectSlotIterator(env, clazz);
-							volatile j9object_t *objectSlotPtr = NULL;
-							while((objectSlotPtr = objectSlotIterator.nextSlot()) != NULL) {
-								didWork |= _markingScheme->markObject(env, *objectSlotPtr);
-							}
-
-							GC_ClassIteratorClassSlots classSlotIterator(clazz);
-							J9Class **classSlotPtr;
-							while((classSlotPtr = classSlotIterator.nextSlot()) != NULL) {
-								didWork |= _markingScheme->markClass(env, *classSlotPtr);
-							}
-						}
-						condYield(env, 0);
-					}
-
-					/* CMVC 131487 */
-					J9HashTableState walkState;
-					/*
-					 * We believe that (NULL == classLoader->classHashTable) is set ONLY for DEAD class loader
-					 * so, if this pointer happend to be NULL at this point let it crash here
-					 */
-					Assert_MM_true(NULL != classLoader->classHashTable);
-					/*
-					 * CMVC 178060 : disable hash table growth to prevent hash table entries from being rehashed during GC yield
-					 * while GC was in the middle of iterating the hash table.
-					 */
-					hashTableSetFlag(classLoader->classHashTable, J9HASH_TABLE_DO_NOT_REHASH);
-					clazz = _javaVM->internalVMFunctions->hashClassTableStartDo(classLoader, &walkState);
-					while (NULL != clazz) {
-						didWork |= _markingScheme->markClass(env, clazz);
-						clazz = _javaVM->internalVMFunctions->hashClassTableNextDo(&walkState);
+	return _staccatoDelegate.allocateAccessBarrier(env);
```

Comment:
We should simplify call of allocateAccessBarrier() in future. There is no need to be so complicate